Using secrets with Kafka Connect Service API (#107)
redpanda-data/cloud-docs: modules/manage/pages/api/cloud-dataplane-api.adoc

=== Create a topic

To create a new Redpanda topic without specifying any further parameters, such as the desired topic-level configuration or partition count, make a POST request to xref:api:ROOT:cloud-api.adoc#post-/v1alpha2/topics[`/v1alpha2/topics`] endpoint:

[,bash]
----
curl -X POST "<dataplane-api-url>/v1alpha2/topics" \
 -H "Authorization: Bearer <token>" \
 -H "accept: application/json" \
 -H "content-type: application/json" \
 -d '{"name":"<topic-name>"}'
----

> === Create a connector

Reviewer: Should this heading be **Create a Kafka Connect connector**, to differentiate from RPCN?

Author: @micheleRP Not sure actually, since I think we refer to "pipelines" in RPCN instead. @asimms41 thoughts?

Reviewer 2: @kbatuigas and @micheleRP, I agree that it would be good to add "Kafka Connect" to each heading, so that if/when users come to this page, they know they are in the wrong place.

> NOTE: Connector management is supported in BYOC and Dedicated clusters only.

Reviewer: Should this say, "Kafka Connect is supported in BYOC and Dedicated clusters only."?

> To create a managed connector, make a POST request to xref:api:ROOT:cloud-api.adoc#post-/v1alpha2/kafka-connect/clusters/-cluster_name-/connectors[`/v1alpha2/kafka-connect/clusters/\{cluster_name}/connectors`]. 

Reviewer: ```suggestion
To create a Kafka Connect connector, make a POST request to xref:api:ROOT:cloud-api.adoc#post-/v1alpha2/kafka-connect/clusters/-cluster_name-/connectors[`/v1alpha2/kafka-connect/clusters/\{cluster_name}/connectors`]. 
```

The following example shows how to create an S3 sink connector with the name `my-connector`:

[,bash]
----
curl -X POST "<dataplane-api-url>/v1alpha2/kafka-connect/clusters/redpanda/connectors" \
 -H "Authorization: Bearer <token>" \
 -H "accept: application/json" \
 -H "content-type: application/json" \
 -d '{"config":{"connector.class":"com.redpanda.kafka.connect.s3.S3SinkConnector","topics":"test-topic","aws.secret.access.key":"${secretsManager:<secret-id>}","aws.s3.bucket.name":"bucket-name","aws.access.key.id":"access-key","aws.s3.bucket.check":"false","region":"us-east-1"},"name":"my-connector"}'
----

[CAUTION]
====

> The field `aws.secret.access.key` in the above example contains sensitive information that usually shouldn't be added to a configuration directly. Use the xref:api:ROOT:cloud-api.adoc#post-/v1alpha2/kafka-connect/clusters/-cluster_name-/secrets[Secrets API] to create a secret that stores the value of the key. You can then use the secret ID to inject the value of the secret in your request.

Reviewer: ```suggestion
The field `aws.secret.access.key` in the example contains sensitive information that usually shouldn't be added to a configuration directly. Use the xref:api:ROOT:cloud-api.adoc#post-/v1alpha2/kafka-connect/clusters/-cluster_name-/secrets[Secrets API] to create a secret that stores the value of the key. You can then use the secret ID to inject the value of the secret in your request.
```

To create a secret that you can reference in the connector configuration request:

[,bash]
----
curl -X POST "https://<dataplane-api-url>/v1alpha2/kafka-connect/clusters/redpanda/secrets" \
 -H 'accept: application/json'\
 -H 'content-type: application/json' \
 -d '{"name":"<connector-name>","secret_data":"<secret-value>"}' 
----

> Use the `id` returned in the Create Secret response to replace the placeholder `<secret-id>` in the Create Connector example above. The syntax `${secretsManager:<secret-id>}` tells the Kafka Connect cluser to load `<secret-id>`. 

Reviewer: ```suggestion
Use the `id` returned in the Create Secret response to replace the placeholder `<secret-id>` in the previous Create Connector example. The syntax `${secretsManager:<secret-id>}` tells the Kafka Connect cluster to load `<secret-id>`. 
```
